crates/common/src/spin_paths.rs

```diff
@@ -0,0 +1,61 @@
+//! Module for resolving various Spin paths
+
+use anyhow::{anyhow, Result};
+use std::path::{Path, PathBuf};
+
+/// The name given to the default manifest file.
+pub const DEFAULT_MANIFEST_FILE: &str = "spin.toml";
+
+/// Resolves a manifest path provided by a user, which may be a file or
+/// directory, to a path to a manifest file.
+pub fn resolve_file_path(provided_path: impl AsRef<Path>) -> Result<PathBuf> {
```

Comment:
This is specific to application manifests.  If we move it, the name needs to change to be manifest-specific.

But... why move it at all?  It fits nicely in the manifest crate.  If you want it for plugins, can't you import the manifest crate and use that?

Reply:
It doesn't live in the manifest crate, it lives in spin-cli.

Moving to spin-manifest is definitely an option, I don't have a strong preference either way.

Reply:
Renamed to `resolve_manifest_file_path`

Reply:
Sorry, should have read other comments first - and apologies for the mistake about the current location.  I take the point about the `manifest` crate being the abstract description of the lockfile, and agree that it would not, in fact, fit there.  Given that, I think you're right that there's no clearer home than `common` (possibly `loader` but that's really concerned with loading, not CLI conventions).

I still have a requested change though, and that is _not_ to combine this with `data_dir`.  There are two separate concerns here:

1. What is the structure of Spin's own on-disk data (registry cache, template store, plugin store) - the internal layout of Spin's private info, abstracted away from the user
2. What are the UI conventions of the Spin CLI (in regard to interpreting paths entered by users) - specific to user interactions

I'd prefer not to mix those concerns into a single "paths" module.  (We could rename the `data_dir` module to something like `layout` over time, but that's not a matter for this PR.)

(also, if I _did_ mix the two, I would call it `paths` not `spin_paths`... we already know it's Spin _grin_)

Apologies again for the mistake and for the back-and-forth.

Reply:
So TL;DR the change you are requesting is to: 

1. Leave `data_dir` unchanged
2. Move `resolve_manifest_file_path` to `paths.rs` 

I'm fine with that.

Reply:
Works for me - thanks!

Reply:
@itowlson done!